java-source/src/main/java/net/corda/training/api/IOUApi.java

```diff
@@ -0,0 +1,203 @@
+package net.corda.training.api;
+
+import net.corda.core.contracts.*;
+import net.corda.core.identity.CordaX500Name;
+import net.corda.core.identity.Party;
+import net.corda.core.messaging.CordaRPCOps;
+import net.corda.core.node.NodeInfo;
+import net.corda.core.transactions.SignedTransaction;
+import net.corda.core.internal.InternalUtils;
+import net.corda.core.internal.FetchDataFlow.Result;
+import net.corda.finance.contracts.GetBalances;
+import net.corda.finance.contracts.asset.Cash;
+import net.corda.finance.flows.CashIssueFlow;
+
+import net.corda.training.flow.IOUIssueFlow;
+import net.corda.training.flow.IOUSettleFlow;
+import net.corda.training.flow.IOUTransferFlow;
+import net.corda.training.flow.SelfIssueCashFlow;
+import net.corda.training.state.IOUState;
+
+import java.util.*;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x500.style.BCStyle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.stream.Collectors;
+
+/**
+ * This API is accessible from /api/iou. The endpoint paths specified below are relative to it.
+ * We've defined a bunch of endpoints to deal with IOUs, cash and the various operations you can perform with them.
+ */
+@Path("iou")
+public class IOUApi{
```

Comment:
We have some inconsistencies around spaces before opening braces - in general we should have a space before all of them